src/backend/u64/field.rs

```diff
@@ -2,6 +2,10 @@
 //! using 64-bit limbs with 128-bit products.
 //! In the 64-bit backend implementation, a `FieldElement` is
 //! represented in radix `2^52`.
+//! 
+//! The basic modular operations have been taken from the 
+//! [Curve25519-dalek repository](https://github.com/dalek-cryptography/curve25519-dalek) and refactored to work 
+//! for the Doppio finite field.
```

Comment:
This can also be sourced from other places so lets speak about citation.

Reply:
I cited the library from where the ops were taken an then refactored and the authors. What else should we add?